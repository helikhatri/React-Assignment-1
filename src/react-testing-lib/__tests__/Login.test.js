import React from 'react';
import Login from "../../containers/Login";
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import LoaderButton from "../../components/LoaderButton";
import Form from "react-bootstrap/Form";
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = mount(<Login />);
});

describe('<Login>', () => {
  it('has a login button', () => {
    expect(wrapper.containsMatchingElement(<LoaderButton type="submit">Login</LoaderButton>)).toBeTruthy();
  });

  it('has a email input field', () => {
    expect(wrapper.containsMatchingElement(<Form.Control type='email' />)).toBeTruthy();
  })

  it('has a password input field', () => {
    expect(wrapper.containsMatchingElement(<Form.Control type='password' />)).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Login>Login</Login>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})