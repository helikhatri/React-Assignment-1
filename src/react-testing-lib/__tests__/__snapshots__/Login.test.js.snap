// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> renders correctly 1`] = `
<div
  className="Login"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
      size="lg"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
      size="lg"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="LoaderButton  btn btn-primary btn-block btn-lg"
      disabled={true}
      title="login"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
